modules/apps/portal-template/portal-template-engine-api/src/main/java/com/liferay/portal/template/engine/TemplateContextHelper.java

```diff
@@ -809,6 +809,10 @@ protected void prepareTiles(
 			(ThemeDisplay)httpServletRequest.getAttribute(
 				WebKeys.THEME_DISPLAY);
 
+		if (themeDisplay == null) {
+			return;
+		}
+
```

Comment:
@jonathanmccann

What is the case that the themeDisplay will be null?

Reply:
The request object retrieved from the execution context does not have a theme display. It seems we have a less populated request object because this is occurring during the notification generation which is outside of a user's direct interaction.

Reply:
@MarinhoFeliphe Did you have any feedback on a way to get a more complete request object? Or should this be forwarded to someone else to review the additional null checks?

Reply:
@jonathanmccann sorry for the late reply,

thanks for the answer, I'm analysing [these failures](https://testray.liferay.com/home/-/testray/case_results?tab=runs&testrayBuildId=2235946411) there are some failed tests that don't seems to be flacky, I'll rebase these changes with master and run these tests again.

Reply:
Thank you, let me know if there's anything I can do.